engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsEventListener.java

```diff
@@ -0,0 +1,335 @@
+        } catch (final IllegalArgumentException | IllegalStateException e) {
+            Log.errorf("unable to extract the export request's UUID from the subject '%s': %s. Original Cloud Event: %s", receivedEvent.getSubject(), e.getMessage(), message.getPayload());
+
+            this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
+
+            return message.ack();
+        }
+
+        // Make sure that we are attempting to handle an export request.
+        if (!this.isAnExportRequest(receivedEvent)) {
+            Log.debugf("[export_request_uuid: %s] ignoring received event from the 'export-requests' topic since either it doesn't come from the 'export-service' or it is not of the 'request-export' type: %s", exportRequestUuid, message.getPayload());
+            return message.ack();
+        }
+
+        // Also, make sure that it contains the expected payload's structure.
+        final Optional<ExportRequest> requestMaybe = receivedEvent.getData(ExportRequest.class);
+        if (requestMaybe.isEmpty()) {
+            Log.errorf("[export_request_uuid: %s] unable to process the export request: the cloud event's data is empty. Original cloud event: %s", exportRequestUuid, message.getPayload());
+
+            this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
+
+            return message.ack();
+        }
+
+        // Extract a few bits of information that will be reused over and over.
+        final ExportRequest request = requestMaybe.get();
+        final ExportRequestClass exportRequest = request.getExportRequest();
+        final String application = exportRequest.getApplication();
+        final UUID resourceUuid = exportRequest.getUUID();
+
+        // If the application target isn't Notifications, then we can simply
+        // skip the message.
+        if (!APPLICATION_NAME.equals(application)) {
+            Log.debugf("[export_request_uuid: %s][resource_uuid: %s] export request ignored for Cloud Event since the target application is '%s': %s", exportRequestUuid, resourceUuid, application, message.getPayload());
+            return message.ack();
+        }
+
+        final String resource = exportRequest.getResource();
+
+        // Check that we support the requested resource type to export.
+        if (!this.isValidResourceType(resource)) {
+            Log.errorf("[export_request_uuid: %s][resource_uuid: %s] export request could not be fulfilled: the requested resource type '%s' is not handled. Original cloud event: %s", exportRequestUuid, resourceUuid, resource, message.getPayload());
+
+            this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
+
+            final ExportError exportError = new ExportError(HttpStatus.SC_BAD_REQUEST, "the specified resource type is unsupported by this application");
+            this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, exportError);
```

Comment:
We could make it obvious in all calls that `exportsService` that `application` is actually a hardcoded value:
```suggestion
            this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, APPLICATION_NAME, resourceUuid, exportError);
```